### DISCUSSION TITLE
[Help]: Update(?) Breaks `pylsp`?: `ModuleNotFoundError: No module named 'pylsp'`

### DISCUSSION THREAD

**Discussion by ghost-in-the-zsh** [mason-org/mason.nvim#1747] [Q&A] [answered]
Hi,

I've been using `nvim`[^1] for several months, including `pylsp`. Everything has been working fine, but over the weekend, I noticed that opening Python files how results in the LSP crashing with the following error message:

From `:LspLog`
```text
[START][2024-07-01 17:32:01] LSP logging initiated
[ERROR][2024-07-01 17:32:01] .../vim/lsp/rpc.lua:770» "rpc"»"/home/<user>/.local/share/nvim/mason/bin/pylsp"»"stderr"» "Traceback (most recent call last):\n  File \"/home/<user>/.local/share/nvim/mason/bin/pylsp\", line 5, in <module>\n from pylsp.__main__ import main\nModuleNotFoundError: No module named 'pylsp'\n"
```

At most, the only thing I did last week was to run the `:Mason` command from inside `nvim` and then upgrade everything that needed upgrading (i.e. `<S-u>`). I don't recall seeing `python-lsp-server` receiving an update at the time, but I could be wrong. Other plugins seem to be working fine still, so the issue seems to be affecting `pylsp` only.

I'd appreciate help and/or guidance. I'm posting here first, b/c:

1. the only thing that stands out to me was using `Mason` to update existing plugins, and
2. I don't have steps to reproduce the problem.

For reference, I'm on Python v3.12.3 and the content of `~/.local/share/nvim/mason/bin` includes:

```text
lrwxrwxrwx 1 <user> <user> 75 May  9 17:42 pylsp -> ~/.local/share/nvim/mason/packages/python-lsp-server/venv/bin/pylsp
```

The sym-link is valid and does show `from pylsp.__main__ import main` on line 5, as noted in the error message. Checking python packages now only shows `python3-lsprotocol` (no more `python-lsp-server`?)

```text
apt search python3-lsp\*
Sorting... Done
Full Text Search... Done
python3-lsprotocol/noble,noble 2023.0.0-1 all
  Python implementation of the Language Server Protocol
```

but installing it makes no difference. Searching the list of issues for `pylsp` and `ModuleNotFoundError` returns unrelated results or nothing respectively.

`:checkhealth mason` shows an error(?):

```text
─────────────────────────────────────────────────────────
mason:

- ERROR No healthcheck found for "mason" plugin.
```

Here are the last few entries for June and today, from `:MasonLog`:
```text
[INFO  Wed 26 Jun 2024 05:49:19 PM MDT] ...e/nvim/lazy/mason.nvim/lua/mason-core/installer/init.lua:245: Installation succeeded for Package(name=mypy)
[INFO  Wed 26 Jun 2024 05:49:22 PM MDT] ...e/nvim/lazy/mason.nvim/lua/mason-core/installer/init.lua:245: Installation succeeded for Package(name=ruff)
[INFO  Mon 01 Jul 2024 04:41:09 PM MDT] ...e/nvim/lazy/mason.nvim/lua/mason-core/installer/init.lua:184: Executing installer for Package(name=ruff) Package(name=ruff)
[INFO  Mon 01 Jul 2024 04:41:09 PM MDT] ...e/nvim/lazy/mason.nvim/lua/mason-core/installer/init.lua:184: Executing installer for Package(name=rust-analyzer) Package(name=rust-analyzer)
[INFO  Mon 01 Jul 2024 04:41:09 PM MDT] ...e/nvim/lazy/mason.nvim/lua/mason-core/installer/init.lua:184: Executing installer for Package(name=clang-format) Package(name=clang-format)
[INFO  Mon 01 Jul 2024 04:41:11 PM MDT] ...e/nvim/lazy/mason.nvim/lua/mason-core/installer/init.lua:245: Installation succeeded for Package(name=rust-analyzer)
[INFO  Mon 01 Jul 2024 04:41:12 PM MDT] ...e/nvim/lazy/mason.nvim/lua/mason-core/installer/init.lua:245: Installation succeeded for Package(name=clang-format)
[INFO  Mon 01 Jul 2024 04:41:13 PM MDT] ...e/nvim/lazy/mason.nvim/lua/mason-core/installer/init.lua:245: Installation succeeded for Package(name=ruff)
```
And what the `:Mason` command shows:

![image](https://github.com/williamboman/mason.nvim/assets/81123101/a6ca74cf-c9ed-4d26-8625-a2cc7d4da9f6)

Thanks in advance for your help in figuring this out,
-me

[^1]: Using the `nvchad v2.5`, to be specific. `nvim` itself has been at version `0.10.0` since my Ubuntu 24.04 LTS upgrade back in April.

**Comment by TamaMcGlinn** [comment, marked as answer]
I just had this issue and fixed it by opening `:Mason`, selecting the (already installed) `python-language-server (pylsp)` and pressing 'u' for update, which seems to have reinstalled the same version.
